### ISSUE TITLE
UnicodeDecodeError's object attribute incorrectly reported unsubscriptable

### ISSUE THREAD

**Issue by igotinfected** [PyCQA/pylint#3639]
### Steps to reproduce
1. catch `UnicodeDecodeError` as `error`
2. try to subscript `error.object`
3. run through pylint

As far as I understood, and going off of the [documenation](https://docs.python.org/3.6/library/exceptions.html#UnicodeError) which states:

> UnicodeError has attributes that describe the encoding or decoding error. For example, **err.object[err.start:err.end]** gives the particular invalid input that the codec failed on.

This code snippet should be all good:

#### poc.py

```python
"""
UnicodeDecodeError: error.object unsubscriptable?
"""

try:
    raise UnicodeDecodeError("utf-8", b'\xe9', 0, 1, "reason")
except UnicodeDecodeError as error:
    print(f"Cannot decode [{str(error.object[error.start : error.end])}]")
    print(dir(error.object))
```

#### output

Note here: `__getitem__` is implemented.
```bash
$ env/bin/python poc.py
Cannot decode [b'\xe9']
['__add__', '__class__', '__contains__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__getnewargs__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__iter__', '__le__', '__len__', '__lt__', '__mod__', '__mul__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__rmod__', '__rmul__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'capitalize', 'center', 'count', 'decode', 'endswith', 'expandtabs', 'find', 'fromhex', 'hex', 'index', 'isalnum', 'isalpha', 'isdigit', 'islower', 'isspace', 'istitle', 'isupper', 'join', 'ljust', 'lower', 'lstrip', 'maketrans', 'partition', 'replace', 'rfind', 'rindex', 'rjust', 'rpartition', 'rsplit', 'rstrip', 'split', 'splitlines', 'startswith', 'strip', 'swapcase', 'title', 'translate', 'upper', 'zfill']
```

However...

### Current behavior

![image](https://user-images.githubusercontent.com/15004223/82567331-8272cb80-9b7d-11ea-9de8-0e64b0d9d746.png)

### Expected behavior

No error report.

### pylint --version output

```bash
$ env/bin/pylint --version
pylint 2.5.2
astroid 2.4.1
Python 3.6.9 (default, Apr 18 2020, 01:56:04)
[GCC 8.4.0]
```

**P.S.:** my use case for this is reporting wrongly encoded characters used for path segments for an API.
